aiida_quantumespresso/workflows/ph/base.py

```diff
@@ -109,6 +109,24 @@ def handle_unrecoverable_failure(self, node):
             self.report_error_handled(node, 'unrecoverable error, aborting...')
             return ProcessHandlerReport(True, self.exit_codes.ERROR_UNRECOVERABLE_FAILURE)
 
+    @process_handler(priority=610, exit_codes=PhCalculation.exit_codes.ERROR_SCHEDULER_OUT_OF_WALLTIME)
+    def handle_scheduler_out_of_walltime(self, node):
+        """Handle `ERROR_SCHEDULER_OUT_OF_WALLTIME` exit code: decrease the max_secondes and restart from scratch."""
+
+        # decrease max_seconds a lot in order to make sure the calculation shut down neatly before reaching the
+        # scheduler wall time
+        factor = 0.5
+        max_seconds = self.ctx.inputs.parameters.get('INPUTPH', {}).get('max_seconds', None)
+        max_seconds_new = max_seconds * factor
```

Comment:
It is possible that you will multiple `None` with `0.5` here. If `max_seconds` is not defined, you should create it yourself from the `max_wallclock_seconds`. I know that currently in the code it is "guaranteed" that it will be set because the workchain itself sets it, but still I think it is more robust to check here explicitly